lib/open_api_spex/cast/all_of.ex

```diff
@@ -0,0 +1,20 @@
+defmodule OpenApiSpex.Cast.AllOf do
+  @moduledoc false
+  alias OpenApiSpex.Cast
+
+  def cast(ctx),
+    do: cast_all_of(ctx, nil)
+
+  defp cast_all_of(%{schema: %{type: _, allOf: [schema | remaining]}} = ctx, result) do
+    with {:ok, value} <- Cast.cast(%{ctx | schema: schema}) do
+      new_schema = Map.put(ctx.schema, :allOf, remaining)
```

Comment:
Can you use `%{ctx.schema, allOf: remaining}` instead? I think it's good rule of thumb is to avoid the `Map` module for structs. `Map.put/3` lacks the protection against adding a key that doesn't belong to a struct. (`Map.replace!/3` has that protection, but it is more verbose and not as commonly used).